azure/services/availabilitysets/client.go

```diff
@@ -64,7 +64,7 @@ func (ac *AzureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.Resou
 	defer done()
 
 	availabilitySet, ok := parameters.(armcompute.AvailabilitySet)
-	if !ok {
+	if !ok && parameters != nil {
```

Comment:
This change is necessary to each service since we first instantiate a `parameters` set to nil in the asyncpoller, only sometimes set it to the result of `Parameters` which has a different concrete type, then pass that value on to this method.

Reflection may be able to work around having to make these changes to the services, but I have a feeling that would end up pretty messy and brittle.